tango_simlib/tests/test_fandango_json_parser.py

```diff
@@ -0,0 +1,196 @@
+                    [cmd_name][INTERESTED_CMD_PROP[1]]]:
+                self.assertEquals(
+                    str(cmd_prop_value).find('Const'), -1,
+                    "The value (%s) of the cmd property, '%s', for the command, '%s',"
+                    "has 'Const'" % (cmd_prop_value, INTERESTED_CMD_PROP[0], cmd_name)
+                    )
+                self.assertEquals(
+                    EXPECTED_TANGO_TYPES[0], type(cmd_prop_value),
+                    "The data type (%s) of the cmd property, '%s', of the command, '%s',"
+                    "is not a tango type" % (cmd_prop_value, INTERESTED_CMD_PROP[0],
+                                             cmd_name)
+                    )
+
+    def test_preprocess_attribute_types(self):
+        """Testing values of attribute properties, 'data_type' and 'data_format',
+        of all attributes are changed to tango types.
+        """
+
+        # testing that data type of specified attr property values is a tango type
+        for attr_name in self.fandango_parser.get_device_attribute_metadata():
+            self.assertEquals(
+                EXPECTED_TANGO_TYPES[0],
+                type(
+                    self.fandango_parser.get_device_attribute_metadata()[attr_name]
+                    [INTERESTED_ATTR_PROP[0]]
+                    ),
+                "The value of attr prop, '%s', of attr name, '%s', is not a tango type."
+                % (INTERESTED_ATTR_PROP[0], attr_name)
+                )
+            self.assertEquals(
+                EXPECTED_TANGO_TYPES[1],
+                type(
+                    self.fandango_parser.get_device_attribute_metadata()
+                    [attr_name][INTERESTED_ATTR_PROP[1]]
+                    ),
+                "The value of attr prop, '%s', of attr name, '%s', is not a tango type."
+                % (INTERESTED_ATTR_PROP[1], attr_name)
+                )
+
+    def test_parsed_attributes(self):
+        """Testing that the attribute information parsed matches with the one captured
+        in the fandango json file.
+        """
+        actual_parsed_attrs = self.fandango_parser.get_device_attribute_metadata()
+        expected_attr_list = [
+            'Status', 'Timing_maximum', 'Timing_average',
+            'Timing_index', 'Timing_info', 'Timing_calls',
+            'State', 'StoredProcedureRelease', 'Timing_minimum'
+            ]
+        actual_parsed_attr_list = actual_parsed_attrs.keys()
+        self.assertGreater(
+            len(actual_parsed_attr_list), 0,
+            "There is no attribute information parsed"
+            )
+        self.assertEquals(
+            expected_attr_list.sort(), actual_parsed_attr_list.sort(),
+            "There are missing attributes"
+            )
+
+        # Using the 'Status' attribute expected results
+        self.assertIn(
+            'Status', actual_parsed_attrs.keys(),
+            "The attribute temperature is not in the parsed attribute list"
```

Comment:
Probably a copy-paste error:   mentions temperature, but should probably mention Status.